pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -89,7 +89,27 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 		sidecarScope := proxy.SidecarScope
 		recomputeOutboundClusters := true
 		if configgen.CanUsePrecomputedCDS(proxy) {
-			if sidecarScope != nil && sidecarScope.XDSOutboundClusters != nil {
+			// if there is no sidecar scope use the one from the gateway
+			// This is a gross hack and can get corrupted quickly as gateways have
+			// different sets of imports. Should be removed. Instead use a default
+			// auto generated sidecar, and use the cache from the sidecar
+			if sidecarScope == nil {
```

Comment:
I doubt if this can happen as we have `SetSidecarScope `called in` initConnectionNode`.

Reply:
ack